spec/controllers/queue_items_controller_spec.rb

```diff
@@ -1,410 +1,333 @@
-      end
+    it_behaves_like "no_current_user_redirect" do
+      let(:action) { post :create, video_id: video.id }
     end
   end
 
   describe "DELETE destroy" do
-    let(:james) { Fabricate(:user) }
-    let (:video1) { Fabricate(:video) }
-    let (:video2) { Fabricate(:video) }
-    let!(:q1) { Fabricate(:queue_item, position: 1, user: pete, video: video1) }
-    let!(:q2) { Fabricate(:queue_item, position: 1, user: james, video: video2) }
-    let!(:q3) { Fabricate(:queue_item, position: 2, user: james, video: Fabricate(:video)) }
-
-    context "when a user is signed in" do
-      before do
-        session[:current_user_id] = james.id
-        request.env['HTTP_REFERER'] = "from_whence_I_came"
-        delete :destroy, id: q2.id
-      end
-
-      context "when current user owns the queue item" do
-        it "deletes a queue item" do
-          expect(QueueItem.count).to eq(2)
-        end
+    context "when current user owns the queue item" do
+      let!(:q1) { Fabricate(:queue_item, position: 1, user: current_user) }
+      let!(:q2) { Fabricate(:queue_item, position: 2, user: current_user) }
+      let!(:q3) { Fabricate(:queue_item, position: 3, user: current_user) }
 
-        it "flashes info message" do
-          expect(flash[:info]).to be_present
-        end
-
-        it "reduces the position of all items deleted item was in front of by one" do
-          expect(q3.reload.position).to eq(1)
-        end
+      before { delete :destroy, id: q2.id }
 
-        it "redirects :back" do
-          expect(response).to redirect_to "from_whence_I_came"
-        end
+      it "deletes a queue item" do
+        expect(QueueItem.count).to eq(2)
       end
 
-      context "when current user does not own queue item" do
-        before { delete :destroy, id: q1.id }
+      it "flashes info message" do
+        expect(flash[:info]).to be_present
+      end
 
-        it "deletes a queue item only if the current user owns it" do
-          expect(:q1).to be_truthy
-        end
+      it "reduces the position of all items deleted item was in front of by one" do
+        expect(q3.reload.position).to eq(2)
+      end
 
-        it "flashes a danger message if no video is deleted" do
-          expect(flash[:danger]).to be_present
-        end
+      it "redirects :back" do
+        expect(response).to redirect_to "from_whence_I_came"
       end
     end
 
-    context "when no user is signed in" do
-      before { delete :destroy, id: q2.id }
+    context "when current user does not own queue item" do
+      let!(:q1) { Fabricate(:queue_item) }
+      let!(:q2) { Fabricate(:queue_item, position: 1, user: current_user) }
+      let!(:q3) { Fabricate(:queue_item, position: 2, user: current_user) }
+    
+      before { delete :destroy, id: q1.id }
 
-      it "doesn't delete any queue items" do
-        expect(QueueItem.count).to eq(3)
+      it "deletes a queue item only if the current user owns it" do
+        expect(:q1).to be_truthy
```

Comment:
I'm not sure this does what you expect. The object exists in memory even if you deleted it from the database in your controller action.